pandas/io/parsers.py

```diff
@@ -1009,6 +999,10 @@ def _create_index(self, ret):
     def get_chunk(self, size=None):
         if size is None:
             size = self.chunksize
+        if self.nrows is not None:
+            if self._currow >= self.nrows:
```

Comment:
raising ``StopIteration`` directly here seems odd. as ``self.read(nrows=0)`` correctly returns the result (as an empty DataFrame)